test/utils/test_negative_sampling.py

```diff
@@ -69,10 +55,6 @@ def test_negative_sampling():
     assert neg_edge_index.size(1) == edge_index.size(1)
     assert is_negative(edge_index, neg_edge_index, (4, 4), bipartite=False)
 
-    neg_edge_index = negative_sampling(edge_index, method='dense')
-    assert neg_edge_index.size(1) == edge_index.size(1)
-    assert is_negative(edge_index, neg_edge_index, (4, 4), bipartite=False)
```

Comment:
Why drop this test?

Reply:
Because the method is now inferred automatically based on the graph size. Since the graph used for the test is small, the method is always dense. This reflects the idea that a sparse method (which is based on a random guessing of the negative edges) is reasonable only when the graph is spare ($E \ll N^2$).

To test both the sparse and the dense method, I added a new test called `test_negative_sampling_with_different_edge_density`. Actually, I think that the whole function `test_negative_sampling` can be removed but I left it there to be sure that the old test still works.